### PR TITLE
Change TALM backend to use selective policy enforcement

### PR DESCRIPTION
**Pull request by Missxiaoguo**
Ref: https://issues.redhat.com/browse/CNF-6505

/cc @jc-rh @sudomakeinstall2

### CODE REVIEW

File: deploy/acm/crds/policy.open-cluster-management.io_placementbindings_crd.yaml [hunk 1]
```diff
@@ -23,6 +27,17 @@ spec:
...
             type: string
+          bindingOverrides:
+            description: BindingOverrides defines the overrides to the Subjects
+            properties:
+              remediationAction:
+                description: This field overrides the policy remediationAction on
+                  target clusters
+                enum:
+                - Enforce
+                - enforce
```
Comment (irinamihai) [inline deploy/acm/crds/policy.open-cluster-management.io_placementbindings_crd.yaml:38]: Is this enum really supposed to be "Enforce, enforce" or should it be "inform, enforce"? :thinking:

Comment (Missxiaoguo) [reply deploy/acm/crds/policy.open-cluster-management.io_placementbindings_crd.yaml:38]: Yeah, it is "Enforce, enforce".  For now, only overriding to enforce is supported.

**Comment by pixelsoccupied** [comment]
Angie wrote in the ticket that this PR is now ready. Can we merge this?

**Comment by Missxiaoguo** [comment]
/unhold

Rebased and re-tested

File: pkg/api/clustergroupupgrades/v1alpha1/types.go [hunk 2]
```diff
@@ -250,8 +250,6 @@ type ClusterGroupUpgradeStatus struct {
 	PlacementBindings []string `json:"placementBindings,omitempty"`
 	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Placement Rules"
 	PlacementRules []string `json:"placementRules,omitempty"`
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Copied Policies"
```
Comment (jc-rh) [inline pkg/api/clustergroupupgrades/v1alpha1/types.go:253]: We can't delete it unless upversion the API. Just mark it as deprecated.

File: controllers/actions.go [hunk 3]
```diff
@@ -133,27 +133,29 @@ func (r *ClusterGroupUpgradeReconciler) deleteClusterLabels(
...
+		}
 	}
-	clusterGroupUpgrade.Status.PlacementRules = nil
 
-	err = utils.DeletePlacementBindings(ctx, r.Client, clusterGroupUpgrade.Namespace, labels)
-	if err != nil {
-		return fmt.Errorf("failed to delete PlacementBindings for CGU %s: %v", clusterGroupUpgrade.Name, err)
-	}
-	clusterGroupUpgrade.Status.PlacementBindings = nil
+	labels := map[string]string{"openshift-cluster-group-upgrades/clusterGroupUpgrade": clusterGroupUpgrade.Name}
```
Comment (jc-rh) [inline controllers/actions.go:143]: What if there are CGUs with the same name in different namespace?

File: pkg/api/clustergroupupgrades/v1alpha1/types.go (hunk 2 above)
Comment (Missxiaoguo) [reply pkg/api/clustergroupupgrades/v1alpha1/types.go:253]: It's a `status` field

File: controllers/actions.go (hunk 3 above)
Comment (jc-rh) [reply controllers/actions.go:143]: FYI, we already have the namespace label for MCV/MCAs

File: pkg/api/clustergroupupgrades/v1alpha1/types.go (hunk 2 above)
Comment (jc-rh) [reply pkg/api/clustergroupupgrades/v1alpha1/types.go:253]: I don't think status fields follow different rules for versioning, do they? They are still part of the API.

**Pull request merged**